js/build.js

```diff
@@ -637,6 +669,56 @@ Fliplet.Widget.instance('chat', function (data) {
     createGroupConversation(groupData);
   }
 
+  function joinPublicChannel() {
+    if (!Fliplet.Navigator.isOnline()){
+      Fliplet.UI.Toast({
+        title: 'You are offline',
+        message: 'An internet connection is necessary to join a public channel.'
+      });
+      return;
+    }
+
+    if (!channelsSelected.length) {
+      return;
+    }
+
+    Fliplet.UI.Toast('Joining channel...');
+    $('.contacts-done-holder').addClass('creating');
+
+    // Add current user to target public channel
+    chat.channels.join(channelsSelected[0].id).then(function (channel) {
+      var toast = Fliplet.UI.Toast('Successfully joined channel');
+
+      // refetch channels next time the view is opened
+      fetchChatChannels = null;
+
+      return getContacts(false).then(function() {
+        return getConversations(false);
+      }).then(function () {
+        chat.poll();
+
+        channelsSelected = [];
+        $('.contacts-done-holder').removeClass('creating');
+        $('[name="group-tabs"]:eq(0)').click();
```

Comment:
Wrap this line to something liek a `switchToDM()` function? Makes it easier if DOM changes.

I wonder if `$('.contacts-done-holder').removeClass('creating');` can be wrapped to a function with a specific purpose too.

Reply:
doen